### PR TITLE
restructure README and add dependencies

### PR DESCRIPTION
**Pull request by MattAtkinson**
Signed-off-by: Matt Atkinson <matt.atkinson@imgtec.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,87 +1,110 @@
...
+#### 3a. Install Dependencies
 
-Linux based applications using OpenWrt can be built by :-
+Before building you need to install dependencies.
 
-    $ make openwrt
+**OpenWrt build dependencies**
 
-OpenWrt binaries can be found at :-
+    $ sudo apt-get install libncurses5-dev libncursesw5-dev zlib1g-dev libssl-dev gawk subversion device-tree-compiler
```
Comment (nikhil-zinjurde-imgtec) [inline README.md]: @MattAtkinson are you sure that for building OpenWrt based Ci40 applications, this subset of dependencies is sufficient out of the total mentioned here https://github.com/CreatorKit/creator-docs/blob/master/GettingStarted.md#installing-dependencies ? Ci40 gateway applications may require some more packages to build, other than those required to build just the vanilla image

File: README.md [hunk 2]
```diff
@@ -1,87 +1,110 @@
...
+
+    $ repo init -u https://github.com/CreatorKit/manifest.git
+    $ repo sync
 
-Enter into build repository and start building :-
+After repo sync is complete, the required repositories are cloned inside the project directory.
+
+Optional: If you wish to build from a tagged release, add a tag name to the repo init command as follows:
+
+     $ repo init -u https://github.com/CreatorKit/manifest.git -b 0.9.5
```
Comment (nikhil-zinjurde-imgtec) [inline README.md]: The correct way of building from a tagged release is mentioned here - https://github.com/CreatorKit/manifest#contents.

And in my opinion, this section of building tagged release is not required in this doc

File: README.md [hunk 3]
```diff
@@ -1,87 +1,110 @@
 ##  Creator-Kit Top level Build system
 
-Creator-Kit requires multiple repositories for building, which are scattered across two GitHub organizations namely CreatorKit, Creatordev, FlowM2M and Cascoda.
+In order to build the IoT Kit projects, we need to collect source code from multiple repositories. The following process will bring these repositories together into a single build environment. The individual repositories used are from these Github organisations:
 
-### Steps for building CreatorKit projects are as follows :-
+* CreatorKit
+* Creatordev
```
Comment (nikhil-zinjurde-imgtec) [inline README.md]: => CreatorDev

File: README.md (hunk 1 above)
Comment (MattAtkinson) [inline README.md]: I tested on a new Ubuntu 16.04 VM using just these dependencies and it has built both OpenWrt and Contiki successfully (I built project 1 "make P=creator-kit-1-cascoda.config"). I'm not sure what the other dependencies are for, maybe they are included within the packages on this list? My only concern is that they may be required for other build options that I have not tried...

**Pull request merged**